parser_test.go

```diff
@@ -104,15 +104,35 @@ EOF
 				&Attribute{
 					Key:      "attr",
 					Value:    hbool(true),
-					Comments: []string{"A comment"},
+					Comments: []string{" A comment"},
```

Comment:
We don't want this, this is backwards breaking but also kind of annoying to deal with for a consumer.

I think normalising comments to a single leading whitespace is probably fine. ie. if there is no whitespace, it will get added. This might require computing the common leading whitespace across the comment block though, to deal with the case of mixed zero and non-zero leading whitespace.